rfcs/2024-03-22-20170-trace-data-model.md

```diff
@@ -0,0 +1,196 @@
+| ParentID | `.["scope_spans"][i]["spans"][j]["parent_span_id"]` |  |
+| Start | `.["scope_spans"][i]["spans"][j]["trace_id"]` |  |
+| Duration | `.["scope_spans"][i]["spans"][j]["trace_id"]` |  |
+| Service | `.["resource"]["attributes"]["service.name"]` |  |
+| Resource | `.["scope_spans"][i]["spans"][j]["name"]` |  |
+| Meta | X | See "Attribute Handling"  |
+| Metrics | `.["scope_spans"][i]["spans"][j]["attributes"][k]` | Metrics[k] is assigned, for Integer or Double type values. |
+
+###### Attribute Handling
+
+For Mapping from Vector to Datadog the scope of ResourceSpan attributes -> ScopeSpan attributes and
+Span attributes are merged. The reverse it will not be expanded.
+
+To begin `.["resource"]["attributes"][*]` are placed in the `Meta` map.
+
+After for any `.["scope_spans"][i]["spans"][j]["attributes"][k]` Meta[k] for any non-Integer and
+non-Double type values. Transcoding the values to Strings.
+
+There are a number of Datadog specific transcodings, but it generally has to do with translating
+semantic conventions to [Datadog conventions][datadog-agent-otlp-ingest-09].
+
+###### Span Name Mapping
+
+Follow [Datadog agent][datadog-agent-otlp-ingest-09] for OTLP -> Datadog. For Datadog -> OTLP Span
+Name.
+
+###### Span Kinds and Types
+
+| Vector Span Kind | Datadog Span Type | Comments |
+|------------------|-------------------|----------|
+| Server | "web" |  |
+| Client | "cache" | If the "db.system" exists in any of the Attribute scopes AND value matches
+"redis" or "memcached" |
+| Client | "db" | If the "db.system" exists in any of the Attribute scopes |
+| Client | "http" | If the above conditions are not true. |
+| * | "custom" |  |
+
+#####
+
+##### From Vector -> Datadog
+
+As `TraceEvent` is a representation of a OTLP `ResourceSpans` message. In Datadog the ingestion
+format expects an Map of Array of Spans. The map key is the `trace_id`.
+
+##### From Datadog -> Vector
+
+There's not a known advantage to providing this.
+
+## Rationale
+
+As of right now there is no alternatives to this, and providing this will allow for interoperability
+between trace formats. Not doing this will prevent the ["must have" OpenTelemetry issue][github-top-level-otel-issue] from being
+able to be completed.
+
+
+## Drawbacks
+
+- TBD
+
+## Prior Art
+
+- [W3C Trace Context](https://www.w3.org/TR/trace-context/) used for propagating data.
+- [OpenTracing Model](https://github.com/opentracing/specification/blob/master/specification.md)
+  predecessor to OpenTelemetry, and used as the basis for Jaeger's internal model.
+- [Zipkin Data Model](https://zipkin.io/zipkin-api/#/default/get_spans)
+- [OpenTelemetry Data Model](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/trace/v1/trace.proto)
+
+## Alternatives
+
+### Using OpenTelemetry-Inspired Model
+
+Rather than needing to respecify the definitions and re-create work done by the wider OpenTelemetry
```

Comment:
Isn't the proposal this? It just does conversions of the proto types to Vector's internal Rust types but keeps the structure of Resource -> InstrumentationScope -> Spans the same?

Reply:
Clarified the title and description.